src/Utils.hs

```diff
@@ -74,12 +74,15 @@ runtimeDir = do
     Right dir -> return dir
     Left e -> error $ T.unpack e
 
-setupNixpkgs :: IO ()
-setupNixpkgs = do
+setupNixpkgs :: Options -> IO ()
+setupNixpkgs o = do
   fp <- getUserCacheDir "nixpkgs"
   exists <- doesDirectoryExist fp
   unless exists $ do
-    _ <- shelly $ run "hub" ["clone", "nixpkgs", T.pack fp] -- requires that user has forked nixpkgs
+    _ <-
+      shelly $ do
```

Comment:
Maybe we need to untangle the modules to be able to use `ourShell`.

Reply:
I'm generally trying to push all the shelly invocations to the lowest level IO functions, and us regular IO for setting up the environment and working directory stuff. Basically, I'm trying to move to only using shelly to directly call a shell command and nothing else fancy.